contests/_22-PDP/c-getout-solution.md

```diff
@@ -0,0 +1,70 @@
+---
+layout: solution
+codename: getout
+---
+
+*Η λυση αυτή απαιτεί γνώσεις BFS και βασικές γνώσεις για bitwise πράξεις.*
+
+## Επεξήγηση εκφώνησης
+
+Μας δίνεται ένας $$N \times N$$ πίνακας με κάποια $$2 \times 1$$ "αυτοκίνητα" που μπορούν να μετακινηθούν οριζοντίως και κάποια $$1 \times 2$$ αυτοκίνητα που μπορούν να μετακινηθούν καθέτως, όσο δεν υπάρχει κάποιο άλλο αυτοκίνητο στον δρόμο τους.
+
+Μας ζητείται να βρούμε τον αριθμό των ελάχιστων κινήσεων ώστε η γραμμή $$R$$ να μείνει ελεύθερη.
+
+## Λύση με αναζήτηση κατά βάθος
+
+Θεωρούμε τον γράφο όπου οι κόμβοι είναι τα δυνατά states του πίνακα, δηλαδή οι πιθανές διατάξεις των αυτοκινήτων. Η ακμή $$S_1 \to S_2$$ υπάρχει ανν  (αν και μόνο αν) στο $$S_1$$ αν κουνήσουμε ένα από τα αυτοκίνητα πηγαίνουμε στο $$S_2$$. Σε αυτόν τον γράφο ξεκινάμε από την αρχική διάταξη $$\textit{init}$$ και στόχος μας είναι να βρούμε τον ελάχιστο αριθμό βημάτων που χρειάζεται για να πάμε σε κάποιο state όπου η γραμμή $$R$$ είναι κενή.
```

Comment:
"Η ακμή $$S_1 \to S_2$$ υπάρχει **ανν** (αν και μόνο αν)" 
1) περισσεύει ένα ν πριν την παρένθεση
2) Το πρώτο "αν" μέσα στην παρένθεση μάλλον δεν χρειάζεται

Reply:
Το άφησα σκέτο "αν και μόνο αν"